0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+    owner = "Target"
+    properties=[
+        PropertyDefinition(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            description="The current state of the catalog product",
+            enum_options=["ACTIVE", "INACTIVE"],
+            required=True
+        ),
+        PropertyDefinition(
+            name="price",
+            data_type=PropertyDefinition.DataType.STRING,
+            description="The price of the catalog product"
+            required=False
+        ),
+        PropertyDefinition(
+            name="return_policy",
+            data_type=PropertyDefinition.DataType.STRING,
+            description="The return policy of the of the catalog product"
+            required=False
+        )])
+```
+
+In order to store a catalog product in global state, we will extend from the 
+definition of a grid product with several fixed fields, namely identifiers, as 
+well as the set of dynamic property values that will conform to the catalog 
+product schema definition.
+
+**Grid Product from the [Product RFC](https://github.com/hyperledger/grid-rfcs/blob/master/text/0005-product.md):**
+
+```
+message Product { 
+    enum ProductNamespace { 
+        UNSET_NAMESPACE = 0; 
+        GS1 = 1; 
+    }
+    
+    ProductNamespace product_namespace = 1; 
+    string product_id = 2; 
+    string owner = 3;
+    repeated PropertyValue properties = 4; 
+}
+```
+
+The catalog_product smart contract would then be responsible for validating the 
+properties field against the CatalogProduct schema at run-time.
+
+A catalog_product entry would reference Grid Product. In addition to 
+the properties defined in the CatalogProduct schema. The data model 
+would look like this:
+```
+CatalogProduct(
+    product_id="gtin", // Reference from Grid Product
+    product_namespace="GS1" // Reference from Grid Product
+    owner='Target', // Reference from Grid Product
+		catalog_id='e70fe351d1d096c'
+    properties=[
+        PropertyValue(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_value=1, # ACTIVE (default)
+        ),
+        PropertyValue(
+            name="price"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="1-10:$50, 10-50:$40, 50+:$30"  # price windows
+        ),
+        PropertyValue(
+            name="return policy"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="30 days from sale"
+        ),
+    ])
+```
+
+## Transaction Payload and Execution 
+
+The following payloads contain an action enum and the associated action payload. 
+This allows for the action payload to be dispatched to the appropriate logic.
+Only the defined actions (enum) are available, and only one Action should be 
+performed in a payload.
+
+### CatalogPayload Transaction
+
+```
+message CatalogPayload { 
```

Comment:
Are each of these payloads going to be handled in different smart contracts? If they are all apart of one it would be better to combine them into on payload. To provide more logical room for each section to have more actions you can separate them by have the actions for catalog start at 1, catalog product at 100  etc.

Reply:
Consolidated into one payload ✅